### ISSUE TITLE
Modify behavior of `output_type="probability"` across TC.

### ISSUE THREAD

**Issue by shantanuchhabra** [apple/turicreate#2667]
Currently, for `output_type="probability"` in a binary classifier setting, all our supervised learning toolkits currently follow the approach of sorting labels lexicographically, and returning the probability of the last label.

This approach is kind of arbitrary, since the labels could be "cat" and "not cat", in which case the user will end up with probabilities for the label, "not cat".

For TC 6.0, we should modify this behavior across Drawing Classifier, Logistic Classifier, SVMs, and other supervised learning toolkits to the following: `output_type="probability"` should only apply if the target values in the data are `0` and `1`.

**Comment by nickjong** [comment]
I don't think we have time to design, develop, and test this change for 6.0. Our other API breaking changes we shipped warnings for in 5.8, and this is a bigger change than any of those.

**Comment by TobyRoseman** [comment]
Perhaps it would be better to just get rid of `output_type="probability"` altogether. It's only supported for binary classification and that has always seemed very strange to me.

Also I agree with @nickjong - any change here is out of scope for 6.0
